src/lib.rs

```diff
@@ -210,18 +227,19 @@ async fn main() {
 
 #[no_mangle]
 pub unsafe extern "C" fn init() {
-    let lottery = Lottery {
+    let mut lottery = Lottery {
         lottery_owner: msg::source(),
         ..Default::default()
     };
 
+    lottery.lottery_state.lottery_owner = lottery.lottery_owner;
```

Comment:
Isn't it the doubling? Why do we need to keep two identical data in `Lottery`?

Reply:
Fixed